snyk/snyk
# fix: fetch patch ignoring command line arguments

BST-889
Fetch patch was making http requests by using needle directly
Changed to use lib request which uses command line arguments when invoking needle
Added tests for lib request
Changed fetch patch fail test to typescript
Removed patch fetch proxy test (now covered by lib request test)
Added @types/sinon devDependency to help write test with sinon assertions and matchers


#### What does this PR do?

Fix issue when snyk protect run as part of snyk wizard and doesn’t keep the --insecure flag
Module fetch-patch was using needle directly
I've changed to use lib/request which handles command line arguments like --insecure

#### How should this be manually tested?

Run `DEBUG=* snyk wizard --insecure` against a project where a patch can be applied. Notice that fetch patch does not use insecure flag and should result in needle using rejectUnauthorized option.

#### What are the relevant tickets?

BST-889

##### Screenshots

<img width="589" alt="Screenshot 2019-10-09 at 22 49 55" src="https://user-images.githubusercontent.com/6598617/66523231-4bb4dd00-eae7-11e9-902e-6a36da0f0992.png">
